core/src/main/java/io/opencensus/tags/TagContexts.java

```diff
@@ -80,7 +78,7 @@ public final TagContextBuilder currentBuilder() {
    * @return an object that defines a scope where the given {@code TagContext} is set to the current
    *     context.
    */
-  public final Scope withTagContext(TagContext tags) {
```

Comment:
Why not final?

Reply:
I overrode this method in TagContextsImpl to ensure that it always puts a TagContextImpl into scope, instead of some other TagContext subclass.

Reply:
Why is not putting other subtypes in scope important? (Curious)

Reply:
I thought that since `TagContextsImpl` is part of a specific OpenCensus implementation that it might be surprising if instance methods like `getCurrentTagContext` could return `TagContext` instances that are not part of that implementation.  TagContextImpl has some invariants that other subclasses might not follow, such as disallowing duplicate tag keys.  Then I made `withTagContext` convert the type to be consistent with `getCurrentTagContext`.

Do you think it would be better to not convert the `TagContext` type?

Reply:
here's a swipe https://github.com/census-instrumentation/opencensus-java/pull/580